src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.InProcOrRemoteHostAnalyzerRunner.cs

```diff
@@ -12,145 +12,219 @@
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Diagnostics.EngineV2;
+using Microsoft.CodeAnalysis.Diagnostics.Telemetry;
 using Microsoft.CodeAnalysis.ErrorReporting;
 using Microsoft.CodeAnalysis.Remote;
 using Microsoft.CodeAnalysis.Shared.TestHooks;
 using Microsoft.CodeAnalysis.Workspaces.Diagnostics;
 using Roslyn.Utilities;
 
-namespace Microsoft.CodeAnalysis.Diagnostics.EngineV2
+namespace Microsoft.CodeAnalysis.Diagnostics
 {
-    internal partial class DiagnosticIncrementalAnalyzer
+    internal class InProcOrRemoteHostAnalyzerRunner
```

Comment:
>InProcOrRemoteHostAnalyzerRunner [](start = 19, length = 32)

Can we separate in-proc from out-of-proc, rather then merging them into a single class?

It seems that the only place that should do analysis in-proc is DefaultDiagnosticAnalyzerService. Perhaps we can move the in-proc part of this class directly to it?

Reply:
This change does not aim at disabling in-proc analysis, only enabling out-of-proc analysis and make it default. Analysis will still run in-proc, if say remote host/OOP dies. Also if we did want to do this eventually, it should be done in a follow-up PR.